docs/upgrade_to_2.1.rst

```diff
@@ -0,0 +1,76 @@
+===========================
+Upgrading to libtorrent 2.1
+===========================
+
+:Author: Arvid Norberg, arvid@libtorrent.org
+
+.. contents:: Table of contents
+  :depth: 2
+  :backlinks: none
+
+In libtorrent 2.1, some parts of the API has changed and some deprecated parts
+have been removed.
+This document summarizes the changes affecting library clients.
+
+C++14 no longer supported
+=========================
+
+libtorrent 2.1 requires at least C++-17. To build with boost build, specify the
+C++ version using the ``cxxstd=17`` build feature (17 is the default).
+
+WebTorrent support
+==================
+
+The major new feature in libtorrent 2.1 is WebTorrent support. WebTorrent
+requires a stun server, it can be configured via the ``webtorrent_stun_server``
+configuration option. It defaults to "stun.l.google.com:19302".
+
+To enable webtorrent support, build with ``webtorrent=on``.
+
+loading and saving torrents
+===========================
+
+The loading and saving of .torrent files and resume data has been simplified and
+responsibilities of classes have become clearer.
+
+	torrent_info
+		Represents the immutable info section of a torrent file. It includes:
+			* the file list
+			* piece hashes (v1 torrents)
+			* merkle root hashes (v2 torrents)
+	add_torrent_params
+		Represents the full torrent file, including:
+			* the torrent_info, which is the immutable part
+			* trackers
+			* web seed URLs
+			* piece layers (v2 torrents)
+
+To load a torrent file from disk, use one of load_torrent_file(),
+load_torrent_buffer() or load_torrent_parsed(). Those functions return an
+add_torrent_params object containing all the information about the torrent.
+
+To save a torrent file from an active torrent session, first ask for the
+add_torrent_params object by calling save_resume_data() and pass in the
+save_info_dict flag. The resulting save_resume_data_alert will contain an
+add_torrent_params object which can be saved to disk using write_torrent_file().
```

Comment:
It would still be nice to be able to serialize/parse torrent_info separately. This is useful for the purposes of storing internal application data to avoid rewriting immutable torrent_info. Do you mind providing such a way?

Reply:
`torrent_handle::torrent_file()` returns the `torrent_info` which can be serialized by `torrent_info::info_section()`. deserializing it can be done via `torrent_info::parse_info_section()`. That last function should probably be a free function and the main way to create a `torrent_info` object (other than parsing a full torrent file).

Reply:
Ok 👍